secretmanager/create_secret.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secretmanager
+
+// [START secretmanager_create_secret]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1beta1"
+	secretspb "google.golang.org/genproto/googleapis/cloud/secrets/v1beta1"
+)
+
+// createSecret creates a new secret with the given name. A secret is a logical
+// wrapper around a collection of secret versions. Secret versions hold the
+// actual secret material.
+func createSecret(w io.Writer, parent, id string) error {
+	// parent := "projects/my-project"
+	// id := "my-secret"
+
+	// Create the client.
+	ctx := context.Background()
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create secretmanager client: %v", err)
+	}
+
+	// Build the request.
+	req := &secretspb.CreateSecretRequest{
+		Parent:   parent,
+		SecretId: id,
+		Secret: &secretspb.Secret{
+			Replication: &secretspb.Replication{
+				Replication: &secretspb.Replication_Automatic_{
+					Automatic: &secretspb.Replication_Automatic{},
```

Comment:
No, and it can't be nil. It explicitly has to be the empty automatic replication policy. For regional replication, there's fields inside the Replication_UserManager struct.